engine/clients/redis/search.py

```diff
@@ -19,27 +20,35 @@
 
 class RedisSearcher(BaseSearcher):
     search_params = {}
-    client = None
+    client: Union[RedisCluster, Redis] = None
     parser = RedisConditionParser()
 
+    knn_conditions: str
+    is_cluster: bool
+    conns: List[Union[RedisCluster, Redis]]
+    search_namespace: RedisSearchIndex
+
     @classmethod
     def init_client(cls, host, distance, connection_params: dict, search_params: dict):
         redis_constructor = RedisCluster if REDIS_CLUSTER else Redis
         cls.client = redis_constructor(
             host=host, port=REDIS_PORT, password=REDIS_AUTH, username=REDIS_USER
         )
         cls.search_params = search_params
-        cls.knn_conditions = "EF_RUNTIME $EF"
-        cls._is_cluster = True if REDIS_CLUSTER else False
+
         # In the case of CLUSTER API enabled we randomly select the starting primary shard
         # when doing the client initialization to evenly distribute the load among the cluster
-        cls.conns = [cls.client]
-        if cls._is_cluster:
+        if REDIS_CLUSTER:
             cls.conns = [
                 cls.client.get_redis_connection(node)
                 for node in cls.client.get_primaries()
             ]
-        cls._ft = cls.conns[random.randint(0, len(cls.conns)) - 1].ft()
+        else:
+            cls.conns = [cls.client]
+
+        cls.is_cluster = REDIS_CLUSTER
+        cls.search_namespace = random.choice(cls.conns).ft()
+        cls.knn_conditions = "EF_RUNTIME $EF"
```

Comment:
btw, if it is a class constant, i think it should be defined on the class level and not in init_client